libraries/wallet/wallet.cpp

```diff
@@ -808,10 +808,25 @@ class wallet_api_impl
          //
          // http://en.wikipedia.org/wiki/Most_vexing_parse
          //
-         fc::ofstream outfile{ fc::path( wallet_filename ) };
+         fc::ofstream outfile{ fc::path( wallet_filename + ".tmp" ) };
          outfile.write( data.c_str(), data.length() );
          outfile.flush();
          outfile.close();
+
+         std::string wallet_file_content;
+         fc::read_file_contents(wallet_filename, wallet_file_content);
+
+         if (wallet_file_content == data) {
+            fc::rename( wallet_filename + ".tmp", wallet_filename );
+         } 
+         else 
+         {
+            disable_umask_protection();
+            throw;
```

Comment:
Please throw an fc::exception subtype with an appropriate error message.

Reply:
Then we also need to throw `fc::exception` subtype in outer try/catch block ? right ?

Reply:
no, the outer catch should re-throw whatever it caught.